scene/2d/node_2d.cpp

```diff
@@ -376,7 +376,10 @@ void Node2D::look_at(const Vector2 &p_pos) {
 
 float Node2D::get_angle_to(const Vector2 &p_pos) const {
 
-	return (get_global_transform().affine_inverse().xform(p_pos)).angle();
+	Vector2 local_pos = to_local(p_pos);
+	local_pos.x *= get_scale().x;
+	local_pos.y *= get_scale().y;
+	return local_pos.angle();
```

Comment:
You should be able to use `return (to_local(p_pos) * get_scale()).angle();`, vectors can be multiplied term-by-term.